### PR TITLE
rsx: Fix surface cache scanning and removal

### PR DESCRIPTION
**Pull request by kd-11**
- Do not restrict range to the small region being sampled. Instead scan the entire range provided. This strategy is more successful at finding overlaps.
- Do bounds check for heading rows appearing outside the range. This really shouldn't happen with the full-range scanning.

Fixes https://github.com/RPCS3/rpcs3/issues/11544

### CODE REVIEW

File: rpcs3/Emu/RSX/Common/surface_store.h [hunk 1]
```diff
@@ -656,6 +656,7 @@ namespace rsx
 				}
 			}
 
+			rsx_log.notice("rsx::surface_cache::check_for_duplicates_fast analysed %u overlapping sections and removed %u", ::size32(sections), removed_count);
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/Common/surface_store.h:659]: maybe only log this if sections and removed are > 0

File: rpcs3/Emu/RSX/Common/surface_store.h [hunk 2]
```diff
@@ -773,11 +776,15 @@ namespace rsx
...
 				if (!valid)
 				{
+					removed_count++;
 					rsx_log.warning("Stale surface at address 0x%x will be deleted", it->base_address);
 					invalidate_surface_address(it->base_address, it->is_depth);
 				}
 			}
 
+			// Notify
+			rsx_log.notice("rsx::surface_cache::check_for_duplicates_fallback analysed %u overlapping sections and removed %u", ::size32(sections), removed_count);
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/Common/surface_store.h:786]: same here

File: rpcs3/Emu/RSX/Common/surface_store.h (hunk 1 above)
Comment (kd-11) [reply rpcs3/Emu/RSX/Common/surface_store.h:659]: This is intentional. The procedure isn't cheap, its good to see when it fails

File: rpcs3/Emu/RSX/Common/surface_store.h (hunk 2 above)
Comment (kd-11) [reply rpcs3/Emu/RSX/Common/surface_store.h:786]: Ditto

**Pull request merged**